### PR TITLE
Fix GitHub issue 24

### LINKED ISSUE #24: Add support for different git ssh users
The plugin is not working for me. I have tracked the problem down to the regex [here](https://github.com/shumphrey/fugitive-gitlab.vim/blob/b8e7b6986c5d13f3e2de2163816af06f74a6f838/autoload/gitlab.vim#L74). The problem is that my GitLab ssh user is `gitlab` not `git`. I know this is not standard and probably means that I set up GitLab incorrectly, but do you think it would be possible to update the regex? I have confirmed that changing it to `...\|git\%(lab\)\?@\|ssh://git\%(lab\)\?\|...` works.

I am also running ssh on a port other than 22. Maybe it is possible to add `let base = substitute(base, ':\d\{1,4}\/', '/', '')` to strip the port before [`let base = tr(base, ':', '/')`](https://github.com/shumphrey/fugitive-gitlab.vim/blob/b8e7b6986c5d13f3e2de2163816af06f74a6f838/autoload/gitlab.vim#L76)?

### PR DESCRIPTION
**Pull request by MarcelRobitaille**
Fix #24.

### CODE REVIEW

**Comment by shumphrey** [comment]
Could we make the ssh user a variable of some sort? Since gitlab is a non-standard user, someone might choose to use a different non standard user also.

something like (not tested)
```
if !exists('g:fugitive_gitlab_ssh_user')
    let g:fugitive_gitlab_ssh_user = 'git'
endif
let base = matchstr(a:remote, '^\%(https\=://\|git://\|' . g:fugitive_gitlab_ssh_user . '@\|ssh://' . g:fugitive_gitlab_ssh_user . '@\)\%(.\{-\}@\)\=\zs\('.domain_pattern.'\)[/:].\{-\}\ze\%(\.git\)\=$')
```

File: autoload/gitlab.vim [hunk 1]
```diff
@@ -67,11 +67,20 @@ function! gitlab#homepage_for_remote(remote) abort
...
     " https://domain/path
     " https://user@domain/path
     " ssh://git@domain/path.git
-    let base = matchstr(a:remote, '^\%(https\=://\|git://\|git@\|ssh://git@\)\%(.\{-\}@\)\=\zs\('.domain_pattern.'\)[/:].\{-\}\ze\%(\.git\)\=$')
+    " ssh://gitlab@domain/path.git
+    " ssh://git@domain:ssh_port/path.git
+    let base = matchstr(a:remote, '^\%(https\=://\|git://\|' . g:fugitive_gitlab_ssh_user . '@\|ssh://' . g:fugitive_gitlab_ssh_user . '@\)\%(.\{-\}@\)\=\zs\('.domain_pattern.'\)[/:].\{-\}\ze\%(\.git\)\=$')
+
+    " Remove port
+    let base = substitute(base, ':\d\{1,4}\/', '/', '')
```
Comment (shumphrey) [inline autoload/gitlab.vim:83]: I think that should be `{1,5}` ports can be up to 65535?

**Comment by shumphrey** [comment]
Added some tweaks and merged

File: autoload/gitlab.vim (hunk 1 above)
Comment (MarcelRobitaille) [reply autoload/gitlab.vim:83]: Good catch.

**Pull request merged**